tests/v3_api/common.py

```diff
@@ -316,4 +328,175 @@ def validate_ingress(p_client, cluster, workload, host, path):
+    return cluster
+
+
+def wait_for_nodes_to_become_active(client, cluster):
+    nodes = client.list_node(clusterId=cluster.id)
+    assert len(nodes) == len(nodes)
+    for node in nodes:
+        node = wait_for_condition(
+            client, node,
+            lambda x: x.state == "active",
+            lambda x: 'State is: ' + x.state,
+            timeout=MACHINE_TIMEOUT)
+
+
+def wait_for_cluster_node_count(client, cluster, expected_node_count, timeout=300):
+    start = time.time()
+    nodes = client.list_node(clusterId=cluster.id)
+    node_count = len(nodes)
+    while node_count != expected_node_count:
+        if time.time() - start > timeout:
+            raise AssertionError("Timed out waiting for state to get to active")
+        time.sleep(.5)
+        nodes = client.list_node(clusterId=cluster.id)
+        node_count = len(nodes)
+
+
+def get_custom_host_registration_cmd(client, cluster, roles, node):
+    allowed_roles = ["etcd", "worker", "controlplane"]
+    cluster_tokens = client.list_cluster_registration_token(clusterId=cluster.id)
+    if len(cluster_tokens) > 0:
+        cluster_token = cluster_tokens[0]
+    else:
+        cluster_token = create_custom_host_registration_token(client, cluster)
+    cmd = cluster_token.nodeCommand
+    for role in roles:
+        assert role in allowed_roles
+        cmd += " --"+role
+    additional_options = " --address " + node.public_ip_address + \
+                         " --internal-address " + node.private_ip_address
+    cmd += additional_options
+    return cmd
+
+
+def create_custom_host_registration_token(client, cluster):
+    cluster_token = client.create_cluster_registration_token(clusterId=cluster.id)
+    cluster_token = client.wait_success(cluster_token)
+    assert cluster_token.state == 'active'
+    return cluster_token
+
+
+def delete_cluster(client, cluster):
+    # Delete Cluster
+    client.delete(cluster)
+    """
+    cluster = wait_for_condition(
+        client, cluster,
+        lambda x: x.state == "removed",
+        lambda x: 'State is: ' + x.state,
+        timeout=m_timeout)
+    assert cluster.state == "removed"
+    """
+
+
+def check_connectivity_between_workloads(p_client1, workload1, p_client2, workload2,
+                                         password, allow_connectivity=True):
+    wl1_pods = p_client1.list_pod(workloadId=workload1.id)
+    wl2_pods = p_client2.list_pod(workloadId=workload2.id)
+    for pod in wl1_pods:
+        for o_pod in wl2_pods:
+            check_connectivity_between_pods(pod, o_pod, password)
+
+
+def check_connectivity_between_workload_pods(p_client, workload, password):
+    pods = p_client.list_pod(workloadId=workload.id)
+    for pod in pods:
+        for o_pod in pods:
+            check_connectivity_between_pods(pod, o_pod, password)
+
+
+def check_connectivity_between_pods(pod1, pod2, password, allow_connectivity=True):
+    pod_ip = pod2.status.podIp
+
+    cmd = "ping -c 1 -W 1 " + pod_ip
```

Comment:
@sangeethah Since this is a new repo, can we switch the image to `leodotcloud/swiss-army-knife`? That we you can actually do a curl instead of just ping? We can add any missing libraries/packages to the above image if you want?

Reply:
@leodotcloud Will get this done in a different PR.